null_int.go

```diff
@@ -0,0 +1,103 @@
+package dosa
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+)
+
+// NullInt is a type that holds an optional int64 value.
```

Comment:
I think we can use https://golang.org/src/database/sql/sql.go?s=4941:5029#L187 instead, which gives us all a lot more for free: Scan(), Value(), etc.